lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -209,6 +178,7 @@ public Dictionary(
     this.needsOutputCleaning = false; // set if we have an OCONV
     flagLookup.add(new BytesRef()); // no flags -> ord 0
 
+    Path tempPath = getDefaultTempDir(); // TODO: make this configurable?
```

Comment:
BTW any idea why the default temp dir is used here for reading from aff file twice, but another temp dir is passed to the constructor and used for similarly-temp-looking task (external dictionary entry sorting)? Can these tasks use the same dir? And if yes, which one? Clients would be simpler if they hadn't to care about temp dirs, but maybe there are use cases for passing the temp dir explicitly.

Reply:
I didn't write this code, so I don't know for sure. I believe the externally-provided temp folder should be used (and reused) but it's of type Directory (it's a Lucene abstraction) so I'm not sure if you'll be able to reuse it for generic Paths.